src/metadata.rs

```diff
@@ -240,7 +245,7 @@ impl Display for Output {
             "\nOutput: {}-{}-{}\n",
             self.name().as_normalized(),
             self.version(),
-            self.build_string()
+            self.build_string().unwrap_or("{build-string-not-set}")
```

Comment:
This is done because Rust has string handling issues, and if you try to println Display then it ignores `fmt::Result` which means a crash from println, and as people throw println in all sorts of places that can cause unexpected crashes, I can't think of a better solution without significant refactoring.

A proper refactor would have two sets of types, the first for parsing YAML to.
The second set is built from the first set with all the hashing, and finalization done, which are then just used for operations.
Hence failing to get finalized dependencies and setting hash would be impossible, and wouldn't require explicit code to handle.